### DISCUSSION TITLE
OpenTofu Migration Guides

### DISCUSSION THREAD

**Discussion by AliSajid** [opentofu/opentofu#813] [General]
Hi,

I am currently the sole person managing the Terraform instrastructure for our research lab. The BUSL issue came right in the middle of an otherwise pleasant cloud migration and while we do not have any major issue right now, we would definitely want to to migrate to something like OpenTofu for both philosophical and legal reasons.

That said, there seems to be a dearth of resources on how to effect such a migration.

Our current stack uses remote runners on Terraform Cloud and uses TFC as the backend to store our state files.

I am trying to map all of that to non-hashicorp alternatives and I would like suggestions from the community. I intend to document my experience as it happens so others may not make the mistakes I would make.

Here's my initial map:

1. terraform binary -> opentofu
2. tfc runners -> github actions

I'm unsure what to use as an alternative backend for terraform state. I would use S3 but securing that could be an issue.

**Comment by alfespa17** [comment]
Maybe you can check Terrakube or OTF as an alternative for TFC.

https://terrakube.org/

https://github.com/leg100/otf

**Comment by avh01** [comment]
GitLab might also be an option they have a registry that can be used to store versioned terraform modules and a http back-end to store the state for each project. They are also going to be releasing an image with Open Tofu in the near future.
https://docs.gitlab.com/ee/user/infrastructure/iac/
There are discussions around releasing a GitLab curated Open Tofu image https://gitlab.com/gitlab-org/terraform-images/-/issues/114#note_1640041444
